src/TypeScript/DefinitionWriter.cpp

```diff
@@ -654,7 +698,7 @@ std::string DefinitionWriter::tsifyType(const Type& type)
             return "string";
         }
         else if (interface.name == "NSDate") {
```

Comment:
Technically, the runtime marshalls NSDate and Date correctly when you pass them to native API, and the right typing for that is Date | NSDate. But then, when you override a method, you have to take care to typecheck and handle both cases for Date and NSDate, this is so annoying. I've talked with @hshristov and agreed to [KISS](https://en.wikipedia.org/wiki/KISS_principle) having it stay just Date.